client/src/main/java/io/pravega/client/tables/impl/MapWrapperImpl.java

```diff
@@ -0,0 +1,668 @@
+            ValueT newValue = toCompute.apply(key, existingValue);
+            if (newValue != null) {
+                if (!areSame(existingValue, newValue)) {
+                    putDirect(key, newValue);
+                }
+            } else {
+                removeDirect(key);
+            }
+            return newValue;
+        }
+        return null;
+    }
+
+    @Override
+    public ValueT computeIfAbsent(@NonNull KeyT key, Function<? super KeyT, ? extends ValueT> toCompute) {
+        ValueT existingValue = get(key);
+        if (existingValue == null) {
+            ValueT newValue = toCompute.apply(key);
+            if (newValue != null) {
+                putDirect(key, newValue);
+            }
+            return newValue;
+        } else {
+            return existingValue;
+        }
+    }
+
+    @Override
+    public Set<KeyT> keySet() {
+        requiresKeyFamily("keySet");
+        return new KeySetImpl();
+    }
+
+    @Override
+    public Collection<ValueT> values() {
+        requiresKeyFamily("values");
+        return new ValuesCollectionImpl();
+    }
+
+    @Override
+    public Set<Entry<KeyT, ValueT>> entrySet() {
+        requiresKeyFamily("entrySet");
+        return new EntrySetImpl();
+    }
+
+    @Override
+    public int size() {
+        requiresKeyFamily("size");
+        long size = keyStream().count();
+        return size > Integer.MAX_VALUE ? Integer.MAX_VALUE : (int) size;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        requiresKeyFamily("isEmpty");
+        return !keyStream().findFirst().isPresent();
+    }
+
+    @Override
+    public void clear() {
+        clear(key -> true);
+    }
+
+    private boolean clear(Predicate<? super KeyT> keyFilter) {
+        requiresKeyFamily("clear");
+        val baseIterator = MapWrapperImpl.this.kvt.keyIterator(this.keyFamily, ITERATOR_BATCH_SIZE, null).asIterator();
+        val deleteFutures = new ArrayList<CompletableFuture<Void>>();
+        while (baseIterator.hasNext()) {
+            val toDelete = baseIterator.next().getItems().stream()
+                    .filter(k -> keyFilter.test(k.getKey()))
+                    .map(k -> TableKey.unversioned(k.getKey()))
+                    .collect(Collectors.toList());
+            if (toDelete.size() > 0) {
+                deleteFutures.add(this.kvt.removeAll(this.keyFamily, toDelete));
+            }
+        }
+
+        if (deleteFutures.size() > 0) {
+            Futures.allOf(deleteFutures).join();
+            return true;
+        }
+
+        return false;
+    }
+
+    //endregion
+
+    //region Helpers
+
+    private void requiresKeyFamily(String opName) {
+        if (this.keyFamily == null) {
+            throw new UnsupportedOperationException(opName + "() requires a Key Family.");
+        }
+    }
+
+    private boolean areSame(ValueT expected, ValueT actual) {
+        return expected == null && actual == null || Objects.equals(expected, actual);
```

Comment:
Objects.equals already handles this null case.

Reply:
Good catch. Fixed it.